Intermittent test db errors

It seems like successfully creating a rethink DB doesn't mean its ready for table creation. The table below exists if I look at the rethink UI. 

This happens everytime I run the tests locally (local docker), so blocks me from running the tests locally.

```
--- FAIL: TestReplayMergeDiffOrdering (0.04s)
panic: gorethink: Database `pachyderm_test_c5b9e90a1920` does not exist. in: 
r.DB("pachyderm_test_c5b9e90a1920").TableCreate("Repos", primary_key="Name") [recovered]
	panic: gorethink: Database `pachyderm_test_c5b9e90a1920` does not exist. in: 
r.DB("pachyderm_test_c5b9e90a1920").TableCreate("Repos", primary_key="Name")
```

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -13,10 +13,9 @@ import (
 	"strings"
 	"time"
 
-
-	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	"github.com/pachyderm/pachyderm/src/client"
 	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	pfsserver "github.com/pachyderm/pachyderm/src/server/pfs"
 	"github.com/pachyderm/pachyderm/src/server/pfs/db/persist"
 	"github.com/pachyderm/pachyderm/src/server/pfs/drive"
@@ -128,40 +127,38 @@ func initDB(session *gorethink.Session, dbName string) error {
 		return nil
 	}
 
-
-// There is a race here.
-    //
-    // When rethink is under load, we consistently fail on the table creation
-    // w a 'db not found' error right after we created it a few lines above.
-    // However, the db shows up fine in the dashboard, so it does in fact get
-    // created. This is only relevant for tests, because thats the only time
-    // we create a bunch of databases while R/W from rethink as well
-    randomTableName := fmt.Sprintf("check_creation_%v", uuid.NewWithoutDashes())
-    randomTableOpts := []gorethink.TableCreateOpts{
-        gorethink.TableCreateOpts{
-            PrimaryKey: "ID",
-        },
-    }
-
-    config := backoff.NewExponentialBackOff()
-    // We want to backoff more aggressively (i.e. wait longer) than the default
-    config.InitialInterval = 1 * time.Second
-    config.Multiplier = 2
-    config.MaxElapsedTime = 5 * time.Minute
-    backoff.RetryNotify(func() error {
-        _, err = gorethink.DB(dbName).TableCreate(randomTableName, randomTableOpts...).RunWrite(session)
-        return err
-    }, config, func(err error, d time.Duration) {
-        fmt.Printf("Error; retrying in %s: %#v\n", d, err)
-    })
-    if err == nil {
-        _, err = gorethink.DB(dbName).TableDrop(randomTableName).RunWrite(session)
-        if err != nil {
-            return err
-        }
-    } else {
-        return fmt.Errorf("Could not validate database (%v) was created.", dbName)
-    }
+	// There is a race here.
+	//
+	// When rethink is under load, we consistently fail on the table creation
+	// w a 'db not found' error right after we created it a few lines above.
+	// However, the db shows up fine in the dashboard, so it does in fact get
+	// created. This is only relevant for tests, because thats the only time
+	// we create a bunch of databases while R/W from rethink as well
+	randomTableName := fmt.Sprintf("check_creation_%v", uuid.NewWithoutDashes())
+	randomTableOpts := []gorethink.TableCreateOpts{
+		gorethink.TableCreateOpts{
+			PrimaryKey: "ID",
+		},
+	}
+	config := backoff.NewExponentialBackOff()
+	// We want to backoff more aggressively (i.e. wait longer) than the default
+	config.InitialInterval = 1 * time.Second
+	config.Multiplier = 2
+	config.MaxElapsedTime = 5 * time.Minute
+	backoff.RetryNotify(func() error {
+		_, err = gorethink.DB(dbName).TableCreate(randomTableName, randomTableOpts...).RunWrite(session)
+		return err
+	}, config, func(err error, d time.Duration) {
+		fmt.Printf("Error; retrying in %s: %#v\n", d, err)
+	})
+	if err == nil {
+		_, err = gorethink.DB(dbName).TableDrop(randomTableName).RunWrite(session)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("Could not validate database (%v) was created.", dbName)
+	}
 
 	// Create tables
 	for _, table := range tables {

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -139,7 +139,7 @@ func initDB(session *gorethink.Session, dbName string) error {
 	config.InitialInterval = 1 * time.Second
 	config.Multiplier = 2
 	config.MaxElapsedTime = 5 * time.Minute
-	tablesToInitialize = make([]string, len(tables))
+	tablesToInitialize := make([]Table, len(tables))
 	copy(tablesToInitialize, tables)
 	backoff.RetryNotify(func() error {
 		// Create tables

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -13,12 +13,15 @@ import (
 	"strings"
 	"time"
 
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	"github.com/pachyderm/pachyderm/src/client"
 	"github.com/pachyderm/pachyderm/src/client/pfs"
 	pfsserver "github.com/pachyderm/pachyderm/src/server/pfs"
 	"github.com/pachyderm/pachyderm/src/server/pfs/db/persist"
 	"github.com/pachyderm/pachyderm/src/server/pfs/drive"
 
+	"github.com/cenkalti/backoff"
 	"github.com/dancannon/gorethink"
 	"github.com/gogo/protobuf/proto"
 	"go.pedge.io/pb/go/google/protobuf"
@@ -124,6 +127,41 @@ func initDB(session *gorethink.Session, dbName string) error {
 		// This function has already run so we abort.
 		return nil
 	}
+
+
+// There is a race here.
+    //
+    // When rethink is under load, we consistently fail on the table creation
+    // w a 'db not found' error right after we created it a few lines above.
+    // However, the db shows up fine in the dashboard, so it does in fact get
+    // created. This is only relevant for tests, because thats the only time
+    // we create a bunch of databases while R/W from rethink as well
+    randomTableName := fmt.Sprintf("check_creation_%v", uuid.NewWithoutDashes())
+    randomTableOpts := []gorethink.TableCreateOpts{
+        gorethink.TableCreateOpts{
+            PrimaryKey: "ID",
+        },
+    }
+
+    config := backoff.NewExponentialBackOff()
+    // We want to backoff more aggressively (i.e. wait longer) than the default
+    config.InitialInterval = 1 * time.Second
+    config.Multiplier = 2
+    config.MaxElapsedTime = 5 * time.Minute
+    backoff.RetryNotify(func() error {
+        _, err = gorethink.DB(dbName).TableCreate(randomTableName, randomTableOpts...).RunWrite(session)
+        return err
+    }, config, func(err error, d time.Duration) {
+        fmt.Printf("Error; retrying in %s: %#v\n", d, err)
+    })
+    if err == nil {
+        _, err = gorethink.DB(dbName).TableDrop(randomTableName).RunWrite(session)
+        if err != nil {
+            return err
+        }
+    } else {
+        return fmt.Errorf("Could not validate database (%v) was created.", dbName)
+    }
 
 	// Create tables
 	for _, table := range tables {

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -140,7 +140,7 @@ func initDB(session *gorethink.Session, dbName string) error {
 	config.Multiplier = 2
 	config.MaxElapsedTime = 5 * time.Minute
 	// Create tables
-	for _, table := range tablesToInitialize {
+	for _, table := range tables {
 		backoff.RetryNotify(func() error {
 			tableCreateOpts := tableToTableCreateOpts[table]
 			_, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session)

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -139,15 +139,16 @@ func initDB(session *gorethink.Session, dbName string) error {
 	config.InitialInterval = 1 * time.Second
 	config.Multiplier = 2
 	config.MaxElapsedTime = 5 * time.Minute
-
+	tablesToInitialize = make([]string, len(tables))
+	copy(tablesToInitialize, tables)
 	backoff.RetryNotify(func() error {
 		// Create tables
-		for i, table := range tables {
+		for i, table := range tablesToInitialize {
 			tableCreateOpts := tableToTableCreateOpts[table]
 			if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
 				return err
 			}
-			tables = append(tables[:i], tables[i+1:]...)
+			tablesToInitialize = append(tablesToInitialize[:i], tablesToInitialize[i+1:]...)
 		}
 		return nil
 	}, config, func(err error, d time.Duration) {

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -25,6 +25,7 @@ import (
 	"go.pedge.io/pb/go/google/protobuf"
 	"go.pedge.io/proto/time"
 	"google.golang.org/grpc"
+	"go.pedge.io/lion"
 )
 
 // A Table is a rethinkdb table name.
@@ -146,7 +147,7 @@ func initDB(session *gorethink.Session, dbName string) error {
 			_, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session)
 			return err
 		}, config, func(err error, d time.Duration) {
-			fmt.Printf("Error; retrying in %s: %#v\n", d, err)
+			lion.Errorf("error creating table %v on database %v; retrying in %s: %v\n", table, dbName, d, err)
 		})
 	}
 

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -15,7 +15,6 @@ import (
 
 	"github.com/pachyderm/pachyderm/src/client"
 	"github.com/pachyderm/pachyderm/src/client/pfs"
-	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	pfsserver "github.com/pachyderm/pachyderm/src/server/pfs"
 	"github.com/pachyderm/pachyderm/src/server/pfs/db/persist"
 	"github.com/pachyderm/pachyderm/src/server/pfs/drive"

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -127,46 +127,33 @@ func initDB(session *gorethink.Session, dbName string) error {
 		return nil
 	}
 
-	// There is a race here.
+	// There is a race here
 	//
 	// When rethink is under load, we consistently fail on the table creation
 	// w a 'db not found' error right after we created it a few lines above.
+	//
 	// However, the db shows up fine in the dashboard, so it does in fact get
 	// created. This is only relevant for tests, because thats the only time
 	// we create a bunch of databases while R/W from rethink as well
-	randomTableName := fmt.Sprintf("check_creation_%v", uuid.NewWithoutDashes())
-	randomTableOpts := []gorethink.TableCreateOpts{
-		gorethink.TableCreateOpts{
-			PrimaryKey: "ID",
-		},
-	}
 	config := backoff.NewExponentialBackOff()
 	// We want to backoff more aggressively (i.e. wait longer) than the default
 	config.InitialInterval = 1 * time.Second
 	config.Multiplier = 2
 	config.MaxElapsedTime = 5 * time.Minute
+
 	backoff.RetryNotify(func() error {
-		_, err = gorethink.DB(dbName).TableCreate(randomTableName, randomTableOpts...).RunWrite(session)
-		return err
+		// Create tables
+		for i, table := range tables {
+			tableCreateOpts := tableToTableCreateOpts[table]
+			if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
+				return err
+			}
+			tables = append(tables[:i], tables[i+1:]...)
+		}
+		return nil
 	}, config, func(err error, d time.Duration) {
 		fmt.Printf("Error; retrying in %s: %#v\n", d, err)
 	})
-	if err == nil {
-		_, err = gorethink.DB(dbName).TableDrop(randomTableName).RunWrite(session)
-		if err != nil {
-			return err
-		}
-	} else {
-		return fmt.Errorf("Could not validate database (%v) was created.", dbName)
-	}
-
-	// Create tables
-	for _, table := range tables {
-		tableCreateOpts := tableToTableCreateOpts[table]
-		if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
-			return err
-		}
-	}
 
 	// Create indexes
 	for _, someIndex := range Indexes {

--- a/src/server/pfs/db/driver.go
+++ b/src/server/pfs/db/driver.go
@@ -139,21 +139,16 @@ func initDB(session *gorethink.Session, dbName string) error {
 	config.InitialInterval = 1 * time.Second
 	config.Multiplier = 2
 	config.MaxElapsedTime = 5 * time.Minute
-	tablesToInitialize := make([]Table, len(tables))
-	copy(tablesToInitialize, tables)
-	backoff.RetryNotify(func() error {
-		// Create tables
-		for i, table := range tablesToInitialize {
+	// Create tables
+	for _, table := range tablesToInitialize {
+		backoff.RetryNotify(func() error {
 			tableCreateOpts := tableToTableCreateOpts[table]
-			if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
-				return err
-			}
-			tablesToInitialize = append(tablesToInitialize[:i], tablesToInitialize[i+1:]...)
-		}
-		return nil
-	}, config, func(err error, d time.Duration) {
-		fmt.Printf("Error; retrying in %s: %#v\n", d, err)
-	})
+			_, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session)
+			return err
+		}, config, func(err error, d time.Duration) {
+			fmt.Printf("Error; retrying in %s: %#v\n", d, err)
+		})
+	}
 
 	// Create indexes
 	for _, someIndex := range Indexes {

